### PR TITLE
Add tools required by bindgen

### PR DESCRIPTION
**Pull request by vireshk**
Add tools required by bindgen, like clang and musl-tools.

Fixes: #55
Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -63,4 +63,13 @@ RUN apt-get -y install debootstrap
...
 RUN apt-get -y --no-install-recommends install shellcheck
 
+# Install bindgen build tools
+RUN apt-get -y --no-install-recommends install llvm-dev libclang-dev clang
+
+# Install musl build tools
+RUN apt-get -y --no-install-recommends install musl-tools
+
+# Install headers required for musl build
+RUN curl https://musl.cc/x86_64-linux-musl-native.tgz | tar -xz  -C /usr/include/
```
Comment (andreeaflorescu) [inline Dockerfile:73]: The docker file is platform independent. It needs to work on aarch64 as well. This will fail on aarch64.

Comment (vireshk) [reply Dockerfile]: Fixed it properly this time among other things.

**Comment by andreeaflorescu** [comment]
I checked the build and it passes on both platforms. If support for musl is needed as well we can merge this from my point of view.

**Comment by vireshk** [comment]
@andreeaflorescu I don't need musl for my work, but the auto builds done for vhost-device need it and makes the pull request unmergeable. So i guess we need it anyway?

**Review by gsserge — APPROVED** [review]

**Review by andreeaflorescu — APPROVED** [review]

**Review by lauralt — APPROVED** [review]

**Pull request merged**